src/platform/plugins/shared/controls/public/actions/create_data_control_panel_action.tsx

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import { apiHasEditorConfig } from '@kbn/control-group-renderer/src/types';
+import type { DataControlState } from '@kbn/controls-schemas';
+import { i18n } from '@kbn/i18n';
+import {
+  apiCanAddNewPanel,
+  apiCanPinPanel,
+  apiIsPresentationContainer,
+} from '@kbn/presentation-containers';
+import { apiPublishesDataViews, type EmbeddableApiContext } from '@kbn/presentation-publishing';
+import { IncompatibleActionError } from '@kbn/ui-actions-plugin/public';
+import type { ActionDefinition } from '@kbn/ui-actions-plugin/public/actions';
+import { openDataControlEditor } from '../controls/data_controls/open_data_control_editor';
+import { ACTION_CREATE_CONTROL, ADD_PANEL_CONTROL_GROUP } from './constants';
+import type { CreateControlTypeContext } from './control_panel_actions';
+import { dataViewsService } from '../services/kibana_services';
+
+let lastUsedDataViewId: string | undefined;
```

Comment:
This is used to remember the last picked dataview ID so that the editor can auto-select it. This used to be handled by the `ControlGroup` but this works well enough.